autotest/ogr/ogr_index_test.py

```diff
@@ -57,225 +48,297 @@ def startup_and_cleanup():
-    gdaltest.p_lyr = gdaltest.p_ds.CreateLayer('index_p')
-
-    ogrtest.quick_create_layer_def(gdaltest.p_lyr, [('PKEY', ogr.OFTInteger)])
-    ogrtest.quick_create_feature(gdaltest.p_lyr, [5], None)
-    ogrtest.quick_create_feature(gdaltest.p_lyr, [10], None)
-    ogrtest.quick_create_feature(gdaltest.p_lyr, [9], None)
-    ogrtest.quick_create_feature(gdaltest.p_lyr, [4], None)
-    ogrtest.quick_create_feature(gdaltest.p_lyr, [3], None)
-    ogrtest.quick_create_feature(gdaltest.p_lyr, [1], None)
-
-# It turns out mapinfo format doesn't allow GetFeatureCount() calls while
-# writing ... it just blows an assert!
-#    if gdaltest.p_lyr.GetFeatureCount() != 7:
-#        gdaltest.post_reason( 'FeatureCount wrong' )
-#        return 'fail'
-
-    # Close and reopen, since it seems the .mif driver does not allow reading
-    # from a newly created (updatable) file.
-
-    gdaltest.p_ds = None
-    gdaltest.p_ds = ogr.OpenShared('index_p.mif', update=0)
-    gdaltest.p_lyr = gdaltest.p_ds.GetLayerByName('index_p')
+    p_ds.Release()
 
-###############################################################################
-# Create a dbf file to be our secondary table.  Close it, and reopen shared.
+    yield
 
+    ogr.GetDriverByName('MapInfo File').DeleteDataSource('index_p.mif')
 
-def test_ogr_index_2():
 
+@contextlib.contextmanager
+def create_join_t_test_file(create_index=False):
     drv = ogr.GetDriverByName('ESRI Shapefile')
-    gdaltest.s_ds = drv.CreateDataSource('join_t.dbf')
-    gdaltest.s_lyr = gdaltest.s_ds.CreateLayer('join_t',
-                                               geom_type=ogr.wkbNone)
+    s_ds = drv.CreateDataSource('join_t.dbf')
+    s_lyr = s_ds.CreateLayer('join_t',
+                             geom_type=ogr.wkbNone)
 
-    ogrtest.quick_create_layer_def(gdaltest.s_lyr,
+    ogrtest.quick_create_layer_def(s_lyr,
                                    [('SKEY', ogr.OFTInteger),
                                     ('VALUE', ogr.OFTString, 16)])
 
     for i in range(20):
-        ogrtest.quick_create_feature(gdaltest.s_lyr, [i, 'Value ' + str(i)], None)
+        ogrtest.quick_create_feature(s_lyr, [i, 'Value ' + str(i)], None)
+
+    if create_index:
+        s_ds.ExecuteSQL('CREATE INDEX ON join_t USING value')
+        s_ds.ExecuteSQL('CREATE INDEX ON join_t USING skey')
 
-    assert gdaltest.s_lyr.GetFeatureCount() == 20, 'FeatureCount wrong'
+    s_ds.Release()
 
-    gdaltest.s_ds.Release()
-    gdaltest.s_lyr = None
-    gdaltest.s_ds = None
+    yield
+
+    ogr.GetDriverByName('ESRI Shapefile').DeleteDataSource('join_t.dbf')
 
-    gdaltest.s_ds = ogr.OpenShared('join_t.dbf', update=1)
-    gdaltest.s_lyr = gdaltest.s_ds.GetLayerByName('join_t')
 
 ###############################################################################
-# Verify a simple join without indexing.
+# Create a MIF file to be our primary table.
+
+
+def test_ogr_index_can_create_MIF_file():
```

Comment:
I'm not convinced that this test and the subsequent one are useful. They're derived from the old test 1 and 2, which may have simply been used as set up functions. Test 1 had no assertions. Test 2 had a single assertion to prove that adding some features to a .dbf worked, but I don't know if that's actually functionality specific to the index-related code here.

Reply:
yes they are mostly setup code. I would just drop them